Actionification

Maciej4 opened the pull request:
This adds a new object, Trigger.java, that should allow for the addition of actions to the parallel scheduler on the press of a button. It also includes two tests.

Maciej4 committed:
fix import error

Maciej4 committed:
fix compile error in intake

Maciej4 committed:
set up for trigger structure

Maciej4 committed:
add more actions

Maciej4 committed:
add tests

HoldenA1 commented on src/main/java/frc/robot/actions/VisionAlign.java:
```diff
@@ -33,10 +32,32 @@
 
     public void loop()
     {
+        //TODO: Combine track cancelling into action
```
There is a lot of commented code here. The main branch should be only working code. If the code is incomplete, it probably shouldn't be merged yet. If this part is being worked on in another branch it should be merged there first.

HoldenA1 commented on src/main/java/frc/robot/util/Trigger.java:
```diff
@@ -0,0 +1,25 @@
+    public void loop()
+    {
+        if(Context.robotController.driverJoystick.getButtonPressed(buttonID)){
+            String seralized = gson.toJson(this.action);
```
It isn't obvious what a google library for converting stuff to JSON is doing here. This a place where comments would be nice to have.

HoldenA1 commented on src/test/java/frc/robot/controllers/DriverJoystickTest.java:
```diff
@@ -0,0 +1,112 @@
+
+    public class Complete extends Action{
+        public void loop()
+        {
```
While code style really isn't that big of an issue, the curly braces are very inconsistent. The opening curly braces should all be on the same line as the function declaration. This should not change for if statements, for loops, while loops, or class declarations. It just has to be consistent.

HoldenA1 reviewed (changes requested):
Bulk of the code looks fine, but some things should be addressed

Maciej4 committed:
fix curly bracket locations

Maciej4 committed:
add comment in gson

The comment explains the use of the GSON library in order to make a deep copy of the action instance.

Maciej4 committed:
remove commented code

Maciej4 commented:
I think that I have fixed all of the issues you have posed.

HoldenA1 reviewed (dismissed):
Looks good now.

Maciej4 committed:
Merge branch 'master' into actionification

Maciej4 commented:
Fixed merge conflicts, which merged into branch and dismissed review. Sorry about that, could you re-review.

HoldenA1 reviewed (approved):
Rip github
